influxdb3_write/src/write_buffer/queryable_buffer.rs

```diff
@@ -0,0 +1,470 @@
+                            batch: chunk.record_batch,
+                            schema: chunk.schema,
+                            timestamp_min_max: chunk.timestamp_min_max,
+                            sort_key: table_buffer.sort_key.clone(),
+                        };
+
+                        persisting_chunks.push(persist_job);
+                    }
+                }
+            }
+
+            persisting_chunks
+        };
+
+        let (sender, receiver) = oneshot::channel();
+
+        let persister = Arc::clone(&self.persister);
+        let executor = Arc::clone(&self.executor);
+        let persisted_files = Arc::clone(&self.persisted_files);
+        let wal_file_number = write.wal_file_number;
+        let buffer = Arc::clone(&self.buffer);
+        let catalog = Arc::clone(&self.catalog);
+
+        tokio::spawn(async move {
+            // persist the catalog
+            loop {
+                let catalog = catalog.clone_inner();
+
+                match persister
+                    .persist_catalog(wal_file_number, Catalog::from_inner(catalog))
+                    .await
+                {
+                    Ok(_) => break,
+                    Err(e) => {
+                        error!(%e, "Error persisting catalog, sleeping and retrying...");
+                        tokio::time::sleep(Duration::from_secs(1)).await;
+                    }
+                }
+            }
+
+            // persist the individual files, building the snapshot as we go
+            let mut persisted_snapshot = PersistedSnapshot::new(wal_file_number);
+            for persist_job in persist_jobs {
+                let path = persist_job.path.to_string();
+                let database_name = Arc::clone(&persist_job.database_name);
+                let table_name = Arc::clone(&persist_job.table_name);
+                let chunk_time = persist_job.chunk_time;
+                let min_time = persist_job.timestamp_min_max.min;
+                let max_time = persist_job.timestamp_min_max.max;
+
+                let (size_bytes, meta) =
+                    sort_dedupe_persist(persist_job, Arc::clone(&persister), Arc::clone(&executor))
+                        .await;
+                persisted_snapshot.add_parquet_file(
+                    database_name,
+                    table_name,
+                    ParquetFile {
+                        path,
+                        size_bytes,
+                        row_count: meta.num_rows as u64,
+                        chunk_time,
+                        min_time,
+                        max_time,
+                    },
+                )
+            }
+
+            // persist the snapshot file
+            loop {
+                match persister.persist_snapshot(&persisted_snapshot).await {
+                    Ok(_) => break,
+                    Err(e) => {
+                        error!(%e, "Error persisting snapshot, sleeping and retrying...");
```

Comment:
At the moment I'm not sure what we can do other than keep trying. If we're unable to persist, it means the object store is unreachable, so writes will then return errors to clients because we'll be able to persist wal files to object storage. So clients attempting to write data will start receiving errors and errors will show up in the logs. It's the equivalent of losing the disk, so not much we can do about it. Unless we decide we want to exit the program entirely.

Reply:
As long as this is backing everything else up including new writes then I think this is fine to keep as is.